packages/astro/src/vite-plugins/astro-swap-functions.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * A plugin for importing Astro's private 'transitions/swap-functions.js' module.
+ * This is temporary solution and can break at any time when end-user's update their `astro` version.
+ */
+import { readFile } from 'node:fs/promises';
+import { createRequire } from 'node:module';
+import { resolve } from 'node:path';
+import type { VitePlugin } from '../types.js';
+
+const require = createRequire(import.meta.url);
+const astroDist = resolve(require.resolve('astro/package.json'), '..');
```

Comment:
Oh interesting, why do we `require.resolve` `astro/package.json` instead of just `astro`? 👀

Reply:
Resolving `astro` gives us the entrypoint of the package. Here we want to instead look exactly for `dist/transitions` directory.

```
> require.resolve("astro")
'/Users/x/tutorialkit/node_modules/.pnpm/astro@4.12.2_@types+node@20.14.11_typescript@5.5.3/node_modules/astro/dist/core/index.js'
> require.resolve("astro/package.json")
'/Users/x/tutorialkit/node_modules/.pnpm/astro@4.12.2_@types+node@20.14.11_typescript@5.5.3/node_modules/astro/package.json'
```

We could of course assume that their entrypoint always stays at `astro/dist/core/index.js`, and traverse the path up from there. 🤔

Reply:
Ah thanks for the explanation!